### DISCUSSION TITLE
Access components inside an STL in a scene

### DISCUSSION THREAD

**Discussion by Y4cine** [zauberzeug/nicegui#1520] [Q&A]
### Question

I am working on an app that shall provide an interactive documentation of an industrial production line.
Several documents describe it. A hierarchical tree (parent/child structure), text snippets for the nodes of the tree, a P&ID, a flowsheet, a 2D-Layout, a 3D-model, etc.
The app is to present 3 views. A fixed one on the left with navigation tree and 2 others displaying whichever from the different documents you want to display side by side.
The actual benefit comes from the synchronisation of the three views. What ever you select in one view, is also selected and highlighted in the other ones. The link being IDs - either identical in the different documents or mapped in tables.

2D drawings will go into SVG components
3D preferably in a scene or if not otherwise doable in a frame with self written javascript (threejs???)

Now the specific question.
Importing an STL worked fine. How do I access the components inside the STL and would you know if STL can transport names and descriptions for its internal components? (I generate them from Solid Works).

Any hint and idea would be very welcome. Thanks in advance.

**Comment by rodja** [comment]
> How do I access the components inside the STL and would you know if STL can transport names and descriptions for its internal components? (I generate them from Solid Works).

This is more an STL/Python question. I suggest you seek help from STL experts on Stack Overflow, Reddit or similar.

**Reply by Y4cine** [reply]
Thank you nonetheless.
